### PR TITLE
Aagarwal/application routing

### PR DESCRIPTION
**Pull request by aagarwal-d2l**

### CODE REVIEW

File: src/components/Framework/Instances/Activity-Instance-users.js [hunk 1]
```diff
@@ -63,10 +66,12 @@ class ActivityInstanceUsers extends EntityMixin(LitElement) {
         </d2l-table-wrapper>`;
 	}
 	handleClick(instance) {
-		const event = new CustomEvent('list-item-clicked-parent', {
-			detail: instance.href
-		});
-		window.dispatchEvent(event);
+		const url = instance.href.split('https://us-east-1.creator-platform.api.dev.brightspace.com/1')[1];
```
Comment (dchopkar) [inline src/components/Framework/Instances/Activity-Instance-users.js:69]: Please create constant for base url

File: src/components/Framework/Instances/Completion-summary/assigned-activity-instance.js [hunk 2]
```diff
@@ -81,10 +84,12 @@ class AssignedActivityInstance extends EntityMixin(LitElement) {
 	}
 
 	handleClick(list) {
-		const event = new CustomEvent('list-item-clicked-parent', {
-			detail: list.href
-		});
-		window.dispatchEvent(event);
+		const url = list.href.split('https://us-east-1.creator-platform.api.dev.brightspace.com/1')[1];
+		redirect(url);
+		// const event = new CustomEvent('list-item-clicked-parent', {
```
Comment (dchopkar) [inline src/components/Framework/Instances/Completion-summary/assigned-activity-instance.js:89]: remove commented code, from all over places

File: src/routes.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+export const loader = () => [
+	{
+		pattern: '/',
+		loader: () => import('./components/Framework/main-activity-list/index'),
+		view: (ctx) => { return html`<main-activity-list-container  token="${ctx.options.token}"></main-activity-list-container>`;}
+	},
+	{
+		pattern: '/activities/:id',
+		loader: () => import('./components/Framework/Activities/index'),
+		view: (ctx) => { return html`<activities-container  token="${ctx.options.token}" href="${`https://us-east-1.creator-platform.api.dev.brightspace.com/1${ctx.path}`}" ></activities-container>`;}
```
Comment (dchopkar) [inline src/routes.js:12]: create constant file for base Url

**Review by dchopkar — CHANGES_REQUESTED** [review]

File: src/components/Framework/Instances/Collections/InstanceTreeViewListItem.js [hunk 4]
```diff
@@ -1,141 +1,140 @@
...
+		}
+		</style>
+		${this.renderItem()}`;
+	}
+	async updated(changedProperties) {
+		super.updated(changedProperties);
+		if (changedProperties.has('_entity')) {
+			if (this._entity) {
+				const self = this;
+				this.instanceHref = this._entity?.getLinkByRel('https://activities.api.brightspace.com/rels/member-instance')?.href;
```
Comment (syedabbasD2L) [inline src/components/Framework/Instances/Collections/InstanceTreeViewListItem.js:74]: use constant from constant file

File: src/components/Framework/Instances/Completion-summary/assigned-activity-instance.js [hunk 5]
```diff
@@ -81,10 +84,12 @@ class AssignedActivityInstance extends EntityMixin(LitElement) {
 	}
 
 	handleClick(list) {
-		const event = new CustomEvent('list-item-clicked-parent', {
-			detail: list.href
-		});
-		window.dispatchEvent(event);
+		const url = list.href.split('https://us-east-1.creator-platform.api.dev.brightspace.com/1')[1];
```
Comment (syedabbasD2L) [inline src/components/Framework/Instances/Completion-summary/assigned-activity-instance.js:87]: use constant from constant file check other places as well

File: src/components/Framework/main-activity-list/index.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import './main-activity-list';
+import { EntityMixin } from '../../../util/EntityStoreMixin';
+class MainActivitiesListIndex extends EntityMixin(LitElement) {
+	static properties = {
+		href: { type: String },
+		token: { type: String },
+	};
+	constructor() {
+		super();
+		this.href = 'https://us-east-1.creator-platform.api.dev.brightspace.com/1/activities';
```
Comment (syedabbasD2L) [inline src/components/Framework/main-activity-list/index.js:11]: use constant from constant file

File: src/components/Framework/main-activity-list/main-activity-list.js [hunk 7]
```diff
@@ -1,64 +1,70 @@
...
+	}
+	async updated() {
+		if (this.href && this.token && !this.isCalled) {
+			this.isCalled = true;
+			this.requestUpdate();
+		}
+	}
+
+	handleClick(e) {
+		const url = e.detail.split('https://us-east-1.creator-platform.api.dev.brightspace.com/1')[1];
```
Comment (syedabbasD2L) [inline src/components/Framework/main-activity-list/main-activity-list.js:60]: use constant from constant file

**Review by dchopkar — APPROVED** [review]

**Pull request merged**